app/scripts/lib/Viewport.js

```diff
@@ -1,244 +1,222 @@
+            linewidth: 0.1
+        });
 
-	container.appendChild( renderer.domElement );
-	renderer.domElement.setAttribute("id", "renderer_canvas");
+        var materialThick = new THREE.LineBasicMaterial({
+            color: 0xeeeeee,
+            linewidth: 2
+        });
 
-	// add subtle ambient lighting
-	var ambientLight = new THREE.AmbientLight(0x555555);
-	scene.add(ambientLight);
+        var line = new THREE.Line(geometry, material, THREE.LinePieces);
+        var lineThick = new THREE.Line(geometryThick, materialThick, THREE.LinePieces);
 
-	// add directional light source
-	var directionalLight = new THREE.DirectionalLight(0xbbbbbb);
-	directionalLight.position.set(50, 30, 50);
-	scene.add(directionalLight);
+        this.scene.add(line);
+        this.scene.add(lineThick);
+    };
 
-	var directionalLight = new THREE.DirectionalLight(0xaaaaaa);
-	directionalLight.position.set(-0.2, -0.8, 1).normalize();
-	scene.add(directionalLight);
+    function onWindowResize() {
 
-	makeGrid();
+        windowHalfX = $container.width() / 2;
+        windowHalfY = $container.height() / 2;
 
-	controls = new THREE.OrbitControls(camera, container);
+        camera.aspect = windowHalfX / windowHalfY;
+        camera.updateProjectionMatrix();
 
-	window.addEventListener( 'resize', onWindowResize, false );
+        renderer.setSize(2 * windowHalfX, 2 * windowHalfY);
 
-	// full screen blur
-	// composer = new THREE.EffectComposer( renderer );
-	// composer.addPass( new THREE.RenderPass( scene, camera ) );
+        this.render();
+    };
 
-	// hblur = new THREE.ShaderPass( THREE.HorizontalBlurShader );
-	// composer.addPass( hblur );
+    function animate() {
 
-	// vblur = new THREE.ShaderPass( THREE.VerticalBlurShader );
-	// // set this shader pass to render to screen so we can see the effects
-	// vblur.renderToScreen = true;
-	// composer.addPass( vblur );
+        requestAnimationFrame(animate.bind(this));
+        render.call(this);
+    };
 
 
-	animate();
+    function render() {
 
-}
+        controls.update();
+        renderer.render(this.scene, camera);
+    };
 
-function makeGrid(){
+    function getBounds(boundingSphere) {
+        return {
+            maxX: boundingSphere.center.x + boundingSphere.radius,
+            minX: boundingSphere.center.x - boundingSphere.radius,
+            maxY: boundingSphere.center.y + boundingSphere.radius,
+            minY: boundingSphere.center.y - boundingSphere.radius,
+            maxZ: boundingSphere.center.z + boundingSphere.radius,
+            minZ: boundingSphere.center.z - boundingSphere.radius
+        };
+    };   
 
-	var l = 60;
+    return function(){
 
-	var axisHelper = new THREE.AxisHelper( l );
-	scene.add( axisHelper );
+        this.scene = null;
 
-	var geometry = new THREE.Geometry();
-	var geometryThick = new THREE.Geometry();
+        this.init = function(){
 
-	var n = l;
-	var inc = 2 * l / n;
-	var rate = 10;
+            container = document.getElementById("viewer");
```

Comment:
I've created a task to explicitly pass the DOM element into this component, rather than using an id like this.  This will help if we do Watch3D.